plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/AmazonS3Reference.java

```diff
@@ -43,21 +42,21 @@
 import java.io.IOException;
 
 /**
- * Handles the shutdown of the wrapped {@link AmazonS3Client} using reference
+ * Handles the shutdown of the wrapped {@link S3Client} using reference
  * counting.
  */
-public class AmazonS3Reference extends RefCountedReleasable<AmazonS3> {
-    AmazonS3Reference(AmazonS3 client) {
+public class AmazonS3Reference extends RefCountedReleasable<S3Client> {
+    AmazonS3Reference(S3Client client) {
         this(client, null);
     }
 
     AmazonS3Reference(AmazonS3WithCredentials client) {
         this(client.client(), client.credentials());
     }
 
-    AmazonS3Reference(AmazonS3 client, @Nullable AWSCredentialsProvider credentials) {
+    AmazonS3Reference(S3Client client, @Nullable AwsCredentialsProvider credentials) {
         super("AWS_S3_CLIENT", client, () -> {
-            client.shutdown();
+            client.close();
             if (credentials instanceof Closeable) {
```

Comment:
Is v2 AwsCredentialsProvider always `Closeable`? In that case I think we may not need this.

Reply:
AwsCredentialsProvider is not always Closeable, although some extension points do implement SdkAutoCloseable. I've changed this check to `instanceof AutoCloseable`.